### PR TITLE
:mortar_board: Topic21 Node relationship definitions

### PR DESCRIPTION
**Pull request by jameshughes89**
### What
Add definitions about how nodes relate to one another. 

### Additional Notes
"Relationships" may not be the best word here.

### CODE REVIEW

File: src/site/topic21.rst [hunk 1]
```diff
@@ -34,6 +34,50 @@ Tree Definition
...
 
+
+Relationships
+-------------
+
+* A **parent/predecessor** of a given node is the node directly above in the hierarchy
+    * Each node can have at most one parent, except for the root, which has no parent
+
+* A **child/successor** of a given node is the node directly below in the hierarchy
+    * Each node can have between zero and many children
```
Comment (twentylemon) [inline src/site/topic21.rst:45]: _between zero and many_ sounds odd. _at least zero_ doesn't work... maybe simply saying nodes can have any number of children? The context kind of implies non-negative and integers.

File: src/site/topic21.rst [hunk 2]
```diff
@@ -34,6 +34,50 @@ Tree Definition
...
+
+* A **descendant** of a given node is the child, or the child's child, or the child's child's child, or ...
+
+
+.. image:: img/tree_base.png
+   :width: 500 px
+   :align: center
+
+* Questions
+    * Can a leaf node have any children?
```
Comment (twentylemon) [inline src/site/topic21.rst:59]: Leaf nodes aren't defined up to this point.

Comment (jameshughes89) [reply src/site/topic21.rst:59]: It will be, #293 content will come before it.

File: src/site/topic21.rst (hunk 1 above)
Comment (jameshughes89) [reply src/site/topic21.rst:45]: agreed.

**Review by twentylemon — APPROVED** [review]

**Pull request merged**